magda-io/magda
# Authorisation for the records (plural) endpoint

### What this PR does

Fixes #2369 

- Adds tests for calling `/records` when there's multiple records with multiple policies in the registry
- Expands capabilities around what the registry OPA parser can understand - in particular policies that use numerics or test for the existence of values
- Adds tests for different kinds of policies in general (e.g. matching on strings, booleans etc)

How to test:

I've made an instance at https://issues-2369-multi-record-metadata-auth.dev.magda.io/. There's 6 records in there - 3 have a `owner_orgunit` policy and 3 have a `public` policy. Right now my user id is listed as the owner of the 3 with the owner_orgunit policy, but if you change that to your user id, you should be able to see that while logged out, https://issues-2369-multi-record-metadata-auth.dev.magda.io/api/v0/registry/records returns the 3 public records and logged in, it returns all 6.

### Checklist
